### DISCUSSION TITLE
Adding a MySQL DB with SSL required, struggling to add the CA.

### DISCUSSION THREAD

**Discussion by ybotmallah** [Budibase/budibase#5515] [Help] [answered]
I am using the DigitalOcean (DO) managed DB service. To access the DB from BudiBase I need to add the CA certificate provided by DO, I have downloaded the .cer file and have opened it in TextEdit, I have then pasted it into SSL key value set with the key ca and the value being the contents of the .cer file (without the begin and end certificate headers). Despite doing this I get the error "self signed certificate in certificate chain". Am I missing something? I applied the same methodology to a Postgres DB also hosted on DO, however with Postgres there is a Ca field.

<img width="1278" alt="image" src="https://user-images.githubusercontent.com/19497751/164331007-7002509c-0a34-4543-998f-fb12f6140bd2.png">

**Comment by melohagan** [comment]
Hi @ybotmallah 

I'm not too familiar with certificates, but this StackOverflow post may help: https://stackoverflow.com/questions/12180552/openssl-error-self-signed-certificate-in-certificate-chain

Seems like there could be an issue with the certificate itself.

**Reply by ybotmallah** [reply]
Hi @melohagan,

Thanks for taking the time to reply to me. I have been referencing #4306 on how to use the SSL feature. Your are correcting in thinking there is a certificate problem, the certificate provided by DigitalOcean is not trusted by a Root CA authority. However by adding the ca key with the value of the certificate provided by Digital Ocean, my understanding is it bypasses the error.

I have tried the exact same setup with a managed Postgres server on Digital Ocean with Budibase successfully. The only difference between the MySQL and Postgres connector in Budibase seems to be the addition of a CA field, which when I paste the certificate provided by DO into when setting up the PG server I don't get the self signed cert error.

<img width="648" alt="image" src="https://user-images.githubusercontent.com/19497751/164543868-8d5bf5a4-3db7-4df5-b274-65850f78f89b.png">

**Comment by melohagan** [comment]
Hey @ybotmallah I see the difference now. 

I had a bit of a dig through the code and it uses **mysql2/promise** which from this example, your setup looks to be correct:
https://github.com/sidorares/node-mysql2/blob/master/examples/ssl/select-over-ssl.js

I believe the difference is that for PostgreSQL you can see that the **RejectUnauthorized** switch is off. In other words, the self signed cert does not throw an error. 

For the **mysql2/promise** library I can see that **RejectUnauthorized** is defaulted to true:
https://github.com/sidorares/node-mysql2/blob/fce1a49b97b259b3926101fd1128835e0aa0bcc4/lib/connection_config.js

This switch should be easy to add to MySQL. I'll create an issue from this discussion.

**Reply by melohagan** [reply]
RejectUnauthorized toggle feature request: https://github.com/Budibase/budibase/issues/5530

Also, this closed issue may add some extra context as well: https://github.com/Budibase/budibase/issues/1967

**Reply by melohagan** [reply]
With that said, you should be able to add the **rejectUnauthorized** param. Try adding this:

![Screenshot 2022-04-22 at 10 10 27](https://user-images.githubusercontent.com/101575380/164675328-bbf216fc-dc65-445a-9744-c076d57456c1.png)

Or if "false" doesn't work then try using **0**

**Reply by ybotmallah** [reply]
@melohagan I've tried with rejectUnauthorized set to both 0 and false with no success, the "self signed certificate in certificate chain" error is still returned.

However you are definitely on the right track as enabling RejectUnauthorized on the PostgreSQL connector and saving, then re-fetching tables gives me the error "Error updating datasource schema". While not explicit in saying it is a connection problem, as this is my assumption as RejectUnauthorized is the only setting toggled between testing.

**Comment by ybotmallah** [comment, marked as answer]
Feature request #5530 which will resolve this issue has been completed and merged into the master, I look forward to testing this in the next release. I will post an update when the next release comes out.
